### PR TITLE
feature/ziggeo-video-recorder: LR-137. Create a separate page for video recorder

### PR DESCRIPTION
**Pull request by mrJmek**
### **WAITING UNTIL #33 WOULD BE MERGED TO PASS CODESHIP TEST**
**Prestory:**
Ця гілка вже була змерджена в 1st demo day, проте .yml так і не запрацювали на продакшені (дякую, за аплодисменти), попередній pull request #15

**Reminder:**
Якщо біля вашого нікнейму є checkbox - !!!ВІТАЮ!!! хоч десь Random.org вказав саме на Вас (=
Будь ласка, перегляньте код, лишіть коментарі, якщо ж зауважень немає - поставте галочку, що все ок.

**Update:**
З усіх змін в порівнянні із попереднім pull request:
- перехід на figaro
- переписаний скріпт з .coffee на .js

**JIRA**
**LR-137 Create a separate page and integrate recorder there:** https://ssu-jira.softserveinc.com/browse/LR-137

- [x] @vovka

> 1) @anatoliiibrahimov
> 2) @AntonMayer
> 3) @brouhaha88

- [x] @OKrain
- [x] @sergiy17

> @LaViki 

History:
> rails g controller video-recorder index
> changed view of video_recorder page
> added Ziggeo modal window recorder to vide_recorder page
> plased all videos playback (players for all recorded videos) on a page
> added success callback (show video token if video status successed) from video recorder througth JS
> rewrote video_recorder_controller with strong parameters
> important tokens (like ziggeo or fb) in separate secrets.yml.ci and secrets.yml files
> all .coffee to .js scripts

### CODE REVIEW

**Comment by sergiy17** [comment]
Everything is ok

**Comment by OKrain** [comment]
Good

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +0,0 @@
-module ApplicationHelper
```
Comment (vovka) [inline app/helpers/application_helper.rb]: Oops, you didn't have to remove ApplicationHelper:)

File: config/secrets.yml [hunk 2]
```diff
@@ -27,3 +33,6 @@ production:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
   facebook_app_id: <%= ENV["FACEBOOK_APP_ID"] %>
   facebook_app_secret: <%= ENV["FACEBOOK_APP_SECRET"] %>
+  ziggeo_api_token: <%= ENV["ZIGGEO_API_TOKEN"] %>
+  ziggeo_private_key: <%= ENV["ZIGGEO_PRIVATE_KEY"] %>
+  ziggeo_encryption_key: <%= ENV["ZIGGEO_ENCRYPTION_KEY"] %>
```
Comment (vovka) [inline config/secrets.yml]: dev/test/prod secrets and env vars looks identical. Can we use yaml inheritance here? like this http://stackoverflow.com/questions/6651275/what-do-the-mean-in-this-database-yml-file

Comment (mrJmek) [inline config/secrets.yml]: This is YAML language syntaxis.
Ok - I'll do that (=
http://rhnh.net/2011/01/31/yaml-tutorial

File: app/assets/javascripts/video_recorder.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+var ready;
+ready = function() {
+
+  $('#reload-page.btn.btn-primary').click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on('submitted', function(data) {
+    var massage, videoStatus;
+    massage = 'The video with token ' +
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l10)

File: app/assets/javascripts/video_recorder.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+var ready;
+ready = function() {
+
+  $('#reload-page.btn.btn-primary').click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on('submitted', function(data) {
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l8)

File: app/assets/javascripts/video_recorder.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+var ready;
+ready = function() {
+
+  $('#reload-page.btn.btn-primary').click(function() {
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l4)

Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l4)

File: app/assets/javascripts/video_recorder.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    massage = 'The video with token ' +
+      data.video.token +
+      ' has been submitted!';
+    videoStatus = $('#video-status');
+    videoStatus.append(massage);
+    videoStatus.show();
+  });
+};
+
+$(document).ready(ready);
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l19)

File: app/assets/javascripts/video_recorder.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  $('#reload-page.btn.btn-primary').click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on('submitted', function(data) {
+    var massage, videoStatus;
+    massage = 'The video with token ' +
+      data.video.token +
+      ' has been submitted!';
+    videoStatus = $('#video-status');
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l13)

File: app/assets/javascripts/video_recorder.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    videoStatus.append(massage);
+    videoStatus.show();
+  });
+};
+
+$(document).ready(ready);
+$(document).on('page:load', ready);
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l20)

Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l20)

File: app/assets/javascripts/video_recorder.js (hunk 4 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["ZiggeoApi" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l8)

File: app/assets/javascripts/video_recorder.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+  $('#reload-page.btn.btn-primary').click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on('submitted', function(data) {
+    var massage, videoStatus;
+    massage = 'The video with token ' +
+      data.video.token +
+      ' has been submitted!';
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/vovka/video_quiz/file/2323527664/issues/source?bid=3205144&fileBranchId=3205156#l12)

File: app/assets/javascripts/video_recorder.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+var ready;
+ready = function() {
+
+  $("#reload-page.btn.btn-primary").click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on("submitted", function(data) {
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["ZiggeoApi" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323728688/issues/source?bid=3205144&fileBranchId=3205156#l8)

File: app/assets/javascripts/video_recorder.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+var ready;
+ready = function() {
+
+  $("#reload-page.btn.btn-primary").click(function() {
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323728688/issues/source?bid=3205144&fileBranchId=3205156#l4)

File: app/assets/javascripts/video_recorder.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    massage = "The video with token " +
+      data.video.token +
+      " has been submitted!";
+    videoStatus = $("#video-status");
+    videoStatus.append(massage);
+    videoStatus.show();
+  });
+};
+
+$(document).ready(ready);
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323728688/issues/source?bid=3205144&fileBranchId=3205156#l19)

File: app/assets/javascripts/video_recorder.js [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  $("#reload-page.btn.btn-primary").click(function() {
+    window.location.reload();
+  });
+
+  return ZiggeoApi.Events.on("submitted", function(data) {
+    var massage, videoStatus;
+    massage = "The video with token " +
+      data.video.token +
+      " has been submitted!";
+    videoStatus = $("#video-status");
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323728688/issues/source?bid=3205144&fileBranchId=3205156#l13)

File: app/assets/javascripts/video_recorder.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    videoStatus.append(massage);
+    videoStatus.show();
+  });
+};
+
+$(document).ready(ready);
+$(document).on("page:load", ready);
```
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2323728688/issues/source?bid=3205144&fileBranchId=3205156#l20)

File: config/secrets.yml (hunk 2 above)
Comment (vovka) [inline config/secrets.yml]: :+1:

**Comment by vovka** [comment]
looks good.gnore codacy

File: app/assets/javascripts/video_recorder.js (hunk 10 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["ZiggeoApi" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2327233888/issues/source?bid=3205144&fileBranchId=3205156#l8)

File: app/assets/javascripts/video_recorder.js (hunk 11 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2327233888/issues/source?bid=3205144&fileBranchId=3205156#l4)

File: app/assets/javascripts/video_recorder.js (hunk 13 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2327233888/issues/source?bid=3205144&fileBranchId=3205156#l13)

File: app/assets/javascripts/video_recorder.js (hunk 12 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2327233888/issues/source?bid=3205144&fileBranchId=3205156#l19)

File: app/assets/javascripts/video_recorder.js (hunk 14 above)
Comment (vovka) [inline app/assets/javascripts/video_recorder.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/vovka/video_quiz/file/2327233888/issues/source?bid=3205144&fileBranchId=3205156#l20)

**Pull request merged**